sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java

```diff
@@ -556,6 +556,10 @@ public void onCallFailed(Exception e) {
         if (isInvalidPartitionWhenDocuments(partition, result)) {
             return result;
         }
+        if (!mNetworkStateHelper.isNetworkConnected()) {
+            completeFutureAndRemovePendingCallWhenDocuments(new DataException("No network detected. List operation is not supported offline."), result);
```

Comment:
should we complete here? I thought the result is resolved in upper level?

Reply:
My understanding is that when you want the async operation to return, you complete it. We know we are not going to do anything else and we have the document instance we are going to return to the user. Why not completing it?